File: oops/polymorphism.py
#polymorphism
#more than one form




#method overloading
#same method different number of arguments

class Maths:
    def add(self):
        print("inside no arg")

    def add(self,num1):
        print("inside one arg")

    def add(self,num1,num2):  #only this works
        print("inside two arg")


m=Maths
m.add(1)


#only recently added works

File: javascript/polymorphism.js
class Calc{
    add(){
        console.log("inside no argument method");
    }
    add(num){
        console.log("inside one argument");
    }
    add(mum1,num2){
        console.log("inside two argumenent");
    }
}

var cal= new Calc()
cal.add(10)
cal.add(10,20)

//only recently added method will works